libs/icu_headroom_metric.py

```diff
@@ -181,6 +181,9 @@ def calculate_icu_utilization_metric(
     metric = current_icu_covid / (icu_data.total_icu_beds - non_covid_patients)
 
     details = can_api_definition.ICUHeadroomMetricDetails(
-        currentIcuCovidMethod=covid_source, currentIcuNonCovidMethod=non_covid_source
+        currentIcuCovidMethod=covid_source,
+        currentIcuCovid=current_icu_covid[current_icu_covid.last_valid_index()],
+        currentIcuNonCovidMethod=non_covid_source,
+        currentIcuNonCovid=non_covid_patients[non_covid_patients.last_valid_index()],
```

Comment:
FWIW- This has been a bigger issue with contact tracing, but sometimes the sub-series that we aggregate together (e.g. contact tracers and new cases / day) into the series that we then graph (tracers hired) have staggered last data points, and so the website copy can end up including a data point that's newer than what was actually used for the metric value.

It would be nice to fix that rather than carry the bug forward. 😁

Reply:
hmm yeah that's a good point.  I guess we would pick the last non null date in the metric - as a nan in the numerator or denominator will cause the result to be nan.